hkmc2/shared/src/main/scala/hkmc2/codegen/Lifter.scala

```diff
@@ -234,55 +236,38 @@ class Lifter(handlerPaths: Opt[HandlerPaths])(using State, Raise):
     val FreeVars(_, cap) = ctx.usedLocals(f.sym)
 
     val fresh = FreshInt()
-
-    val varsMap = cap.map: s =>
-        val id = fresh.make
-        val nme = s.nme + id + "$"
-        val varSym = VarSymbol(Tree.Ident(nme))
-        val fldSym = BlockMemberSymbol(nme, Nil)
-        val fldDef = TermDefinition(
-          S(clsSym),
-          syntax.ImmutVal,
-          fldSym,
-          Nil, N, N,
-          S(Term.Ref(s)(Tree.Ident(s.nme), 666)), // FIXME: 666 is a dummy value
-          FlowSymbol("‹class-param-res›"),
-          TermDefFlags.empty,
-          Nil
-        )
-        fldSym.defn = S(fldDef)
-        s -> (
-          varSym,
-          fldDef,
-        )
-      .toMap
-
-    val varsList = cap.toList
+    
+    val sortedVars = cap.toArray.sortBy(_.uid).map: sym =>
+      val id = fresh.make
+      val nme = sym.nme + id + "$"
+      
+      val varSym = VarSymbol(Tree.Ident(nme))
+      val fldSym = BlockMemberSymbol(nme, Nil)
+      
+      val p = Param(FldFlags.empty.copy(value = true), varSym, None)
+      varSym.decl = S(p) // * Currently this is only accessed to create the class' toString method
+      
+      val vd = ValDefn(
+        S(clsSym),
+        syntax.ImmutVal,
+        fldSym,
+        Value.Ref(varSym)
+      )
+      
+      (sym -> varSym, p, vd)
```

Comment:
I now compute all the relevant things in one place that's much easier to inspect and understand.